rllib/env/vector_env.py

```diff
@@ -232,11 +257,38 @@ def reset_at(self, index: Optional[int] = None) -> EnvObsType:
             index = 0
         return self.envs[index].reset()
 
+    @override(VectorEnv)
+    def restart_at(self, index: Optional[int] = None) -> None:
+        if index is None:
+            index = 0
+        old_env = self.envs[index]
+        # Re-create the sub-env at the new index.
+        self.envs[index] = self.make_env(index)
+        # Try closing down the old (possibly faulty) sub-env, but ignore errors.
+        try:
+            old_env.close()
+        except Exception as e:
+            if log_once("close_sub_env"):
+                logger.warning(
+                    "Trying to close old and replaced sub-environment (at vector "
+                    f"index={index}), but closing resulted in error:\n{e}"
+                )
```

Comment:
why not close first, then try to re-make the env?
then you don't need to remember the old_env?

```
try:
    self.envs[index].close()
except Exception as e:
    logger.warning()

self.envs[index] = self.make_env(index)
```